agent/agent.go

```diff
@@ -343,6 +343,28 @@ func (a *Agent) processNextTask(ctx context.Context, nt *apimodels.NextTaskRespo
 	if nt.ShouldTeardownGroup {
 		// Tear down the task group if the task group is finished.
 		a.runTeardownGroupCommands(ctx, tc)
+
+		// Terminate host if a task groups is complete in a single host distro.
+		if a.opts.SingleTaskDistro {
+			err := a.comm.DisableHost(ctx, a.opts.HostID, apimodels.DisableInfo{Reason: "Single task distro ran a task"})
+			if err != nil {
+				span.RecordError(err, trace.WithAttributes(attribute.String("task.id", tc.task.ID)), trace.WithStackTrace(true))
+				grip.Critical(message.WrapError(err, message.Fields{
+					"message":    "error disabling host after task group completion",
+					"task":       tc.task.ID,
+					"host":       a.opts.HostID,
+					"task_group": nt.TaskGroup,
+				}))
+
+				return processNextResponse{
```

Comment:
Do we need this return? It looks duplicated like L368 which will get called without this return here.

Reply:
we return the "err" here and I could take out the err of the if statement and combine it but I didn't think it was that important haha

Reply:
Ah I see it! Do you think it would be easier to read if we established err outside the ifs and modified the below return?

Also, both span.RecordError and message.WrapError no-op when err is nil, so we don't need to test if it is nil before doing them